### ISSUE TITLE
[Desktop] generic YT favicon displayed for verified creator in certain cases

### ISSUE THREAD

**Issue by LaurenWags** [brave/brave-browser#11883]
## Description 

Found while testing https://github.com/brave/brave-browser/issues/9133

Sometimes the generic YT favicon is shown for a YT publisher instead of the user's actual favicon. So far I've only observed this when visiting a "User" URL before a "channel" URL or a specific video. When "user" URL is visited first, then the generic favicon is always shown.

Visiting the "channel" URL or a specific video shows the expected favicon *if* the "user" URL wasn't visited first.

## Steps to Reproduce

   1. Clean profile, enable Rewards (prod). Wait for all information (publisher indexes, etc) to be downloaded.
   2. Visit youtube.com/user/JJDSampson
   3. Once page loads, verified icon id shown on BAT logo
   4. Click on BAT logo.

## Actual result:

Generic YT favicon is shown

<img width="1197" alt="Sampson generic YT favicon 1 16 x" src="https://user-images.githubusercontent.com/28145373/94436997-2f340980-016b-11eb-9a32-0781720b0734.png">

## Expected result:
Should show user's actual favicon as it does in 1.14.x:
<img width="1198" alt="Sampson defined YT favicon 1 14 x" src="https://user-images.githubusercontent.com/28145373/94437069-44109d00-016b-11eb-82b2-b1384c1b4c38.png">

## Reproduces how often: 

easily

## Brave version (brave://version info)

Brave | 1.16.42 Chromium: 86.0.4240.55 (Official Build) nightly (x86_64)
-- | --
Revision | a6d625ef6f7fe8ea0675f1cf759155a05ee1be40-refs/branch-heads/4240@{#953}
OS | macOS Version 10.14.6 (Build 18G3020)

## Version/Channel Information:

- Can you reproduce this issue with the current release? no, not with 1.14.x
- Can you reproduce this issue with the beta channel? no, not with 1.15.x
- Can you reproduce this issue with the nightly channel? yes with 1.16.x

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? 
- Does the issue resolve itself when disabling Brave Rewards? 
- Is the issue reproducible on the latest version of Chrome? 

## Miscellaneous Information:

If you visit the "channel" or video URL before visiting the "user" URL, you will get the expected favicon. Each check below was done with a clean profile:

Channel - https://www.youtube.com/channel/UCD7siefto4I33L4U61IU5LQ
<img width="1200" alt="Sampson Channel YT 1 16 x" src="https://user-images.githubusercontent.com/28145373/94437463-c731f300-016b-11eb-8603-b11718b55add.png">

Video - https://www.youtube.com/watch?v=uWKCm2GKVyE
<img width="1198" alt="Sampson video YT 1 16 x" src="https://user-images.githubusercontent.com/28145373/94437471-c8632000-016b-11eb-88a4-d195ad3d14e7.png">

cc @NejcZdovc @emerick @brave/legacy_qa

**Comment by emerick** [comment]
Fix is in version `1.0.28` of `Brave Local Data Updater` (see brave://components).

**Comment by LaurenWags** [comment]
Verified passed with

```
Brave | 1.16.47 Chromium: 86.0.4240.55 (Official Build) dev (x86_64)
-- | --
Revision | a6d625ef6f7fe8ea0675f1cf759155a05ee1be40-refs/branch-heads/4240@{#953}
OS | macOS Version 10.14.6 (Build 18G3020)
```

* Verified STR from description and component version from https://github.com/brave/brave-browser/issues/11883#issuecomment-700760522

<img width="1197" alt="Screen Shot 2020-09-30 at 11 21 38 AM" src="https://user-images.githubusercontent.com/28145373/94705528-29772900-030f-11eb-8680-f0d1bff88621.png">

---------
